### PR TITLE
DAOS-4946 test: Fix the offline test cases

### PR DESCRIPTION
**Pull request by rpadma2**
Summary: Fix the offline test cases to use data.
Address the pool_destroy issue stated under DAOS-4946.
Start moving common methods to osa_utils.py

Signed-off-by: Padmanabhan <ravindran.padmanabhan@intel.com>

### CODE REVIEW

File: src/tests/ftest/osa/osa_offline_parallel_test.py [hunk 1]
```diff
@@ -252,10 +177,5 @@ def test_osa_offline_parallel_test(self):
...
         """
-        # Perform drain testing with 1 to 2 pools
-        # Two pool testing blocked by DAOS-5333.
-        # Fix range from 1,2 to 1,3
-        self.run_offline_parallel_test(1)
-        # Perform drain testing : inserting data in pool
-        # Bug : DAOS-4946 blocks the following test case.
-        # self.run_offline_parallel_test(1, True)
+        # Run the parallel offline test.
+        self.run_offline_parallel_test(1, True)
```
Comment (shimizukko) [inline src/tests/ftest/osa/osa_offline_parallel_test.py:181]: Do we need to try the range from 1 to 3 like we did in osa_offline_drain.py?

**Review by shimizukko — APPROVED** [review]
LGTM

File: src/tests/ftest/osa/osa_offline_parallel_test.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_parallel_test.py:181]: Presently, I don't want to try it... I may need some fine tuning in the script. It causes issue now. Will try it later.

**Review by amjustin13 — APPROVED** [review]
LGTM

**Comment by rpadma2** [comment]
Daos Core tests has following bug tickets: DAOS-5873y, DAOS-5884, DAOS-5922, DAOS-5882 and DAOS-5927.

**Comment by rpadma2** [comment]
mdtest_small failure: DAOS-6012 created.

**Comment by rpadma2** [comment]
All the changes done are only for OSA related tests. None of the failures are related to the PR changes. PR ready for gatekeeper to merge.

**Pull request merged**